net_handler/net_handler.c

```diff
@@ -156,18 +157,20 @@ int main ()
 		}
 		cli_addr_len = sizeof(struct sockaddr_in);
 		log_printf(DEBUG, "Received connection request from %s:%d", inet_ntoa(cli_addr.sin_addr), ntohs(cli_addr.sin_port));
-		uint8_t is_dawn;
-		if (read(connfd, &is_dawn, 1) == -1) {
+		
+		//get the client ID (first byte on the socket from client)
+		if (read(connfd, &client_id, 1) == -1) {
 			log_printf(ERROR, "Couldn't get client type: %s", strerror(errno));
 			continue;
 		}
-		//if the incoming request is dawn or shepherd, start the appropriate threads
-		if (is_dawn && robot_desc_read(DAWN) == DISCONNECTED) {
-			log_printf(DEBUG, "Starting Dawn connection");
-			start_tcp_conn(DAWN, connfd, 1);
-		} else if (robot_desc_read(SHEPHERD) == DISCONNECTED) {
+		
+		//if the incoming request is shepherd or dawn, start the appropriate threads
+		if (client_id == 0 && robot_desc_read(SHEPHERD) == DISCONNECTED) {
```

Comment:
Not sure if you want this, but I was thinking about also checking the port here as the ports are something that are easily binded by both sides. Adds a little bit of extra security. Either way though, we should delete the two functions `is_dawn` and `is_shepherd` from this file.

Reply:
👌

Reply:
Actually, I'm going to keep `is_dawn` and `is_shepherd` because we still have 3 conditions for starting the thread, it's just different from the original 3 conditions. Our new conditions are:
Is `client_id` valid?

Reply:
Sure but theres no need for 2 extra functions that only get used in 1 place. Just makes the code unnecessarily complex

Reply:
Fixed it